File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        output = []
        l, r = 0, k

        while r <= len(nums):
            output.append(max(nums[l:r]))
            l += 1
            r += 1

        return output

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        const productArr = [];

        let prefixProduct = 1;
        for(let i = 0; i < nums.length; i++) {
            productArr[i] = prefixProduct; 
            prefixProduct *= nums[i];
        }

        console.log(productArr);

        let postfixProduct = 1;
        for(let i = productArr.length - 2; i >= 0; i--) {
            postfixProduct *= nums[i + 1];
            productArr[i] *= postfixProduct;
        }

        return productArr;
    }
}
